misc/python/materialize/checks/range.py

```diff
@@ -0,0 +1,110 @@
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0.
+from textwrap import dedent
+from typing import List
+
+from materialize.checks.actions import Testdrive
+from materialize.checks.checks import Check
+
+
+class Range(Check):
+
+    def initialize(self) -> Testdrive:
+        return Testdrive(
+            dedent(
+                """
+            > DROP MATERIALIZED VIEW IF EXISTS int_range_view1;
+            > DROP MATERIALIZED VIEW IF EXISTS int_range_view2;
+            > DROP TABLE IF EXISTS int_range_table;
+            > CREATE TABLE int_range_table (f1 INT, f2 INT8RANGE);
+            > INSERT INTO int_range_table VALUES (1, '[2,16]'::INT8RANGE);
+            > INSERT INTO int_range_table VALUES (2, NULL);
+            """
+            )
+        )
+
+    def manipulate(self) -> List[Testdrive]:
+        return [
+            Testdrive(dedent(s))
+            for s in [
+                """
+                > CREATE MATERIALIZED VIEW int_range_view1 AS SELECT f1, f2 FROM int_range_table;
+                > INSERT INTO int_range_table VALUES (3, '[2,6)'::INT8RANGE);
+                > UPDATE int_range_table SET f2 = '[2,2)'::INT8RANGE WHERE f1 = 3;
+            """,
+                """
+                > CREATE MATERIALIZED VIEW int_range_view2 AS SELECT f1, f2 FROM int_range_table;
+                > INSERT INTO int_range_table VALUES (4, '[2,20]'::INT8RANGE);
+            """,
+            ]
+        ]
+
+    def validate(self) -> Testdrive:
+        return Testdrive(
+            dedent(
+                """
+            > SELECT * FROM int_range_table ORDER BY f1 ASC;
+            1 [2,17)
+            2 <null>
+            3 empty
+            4 [2,21)
+                        
+            > SELECT * FROM int_range_view1 ORDER BY f1 ASC;
+            1 [2,17)
+            2 <null>
+            3 empty
+            4 [2,21)
+
+            > SELECT * FROM int_range_view2 ORDER BY f1 ASC;
+            1 [2,17)
+            2 <null>
+            3 empty
+            4 [2,21)
+
+            > SELECT f1, COALESCE(f2, '[10,80)'::INT8RANGE) from int_range_view1 ORDER BY f1 ASC;
+            1 [2,17)
+            2 [10,80)
+            3 empty
+            4 [2,21)
+
+            > SELECT f1, GREATEST(f2, '[2,5]'::INT8RANGE), LEAST(f2, '[2,5]'::INT8RANGE) from int_range_view1 ORDER BY f1 ASC;
+            1 [2,17) [2,6)
+            2 [2,6) [2,6)
+            3 [2,6) empty
+            4 [2,21) [2,6)
+
+            > SELECT f1, f2, NULLIF(f2, '[2,17)'::INT8RANGE) from int_range_view1 ORDER BY f1 ASC;
+            1 [2,17) <null>
+            2 <null> <null>
+            3 empty empty
+            4 [2,21) [2,21)
+
+            > SELECT f1, f2, LOWER(f2), UPPER(f2), ISEMPTY(f2), LOWER_INC(f2), UPPER_INC(f2), LOWER_INF(f2), UPPER_INF(f2) from int_range_view1 ORDER BY f1 ASC;
```

Comment:
These functions are available in Materialize but not documented in https://materialize.com/docs/sql/functions/. Shall I create a docs ticket?

Reply:
Please do!